pool/app/session/repo/repo.ml

```diff
@@ -844,12 +850,49 @@ module Sql = struct
       | "" -> None
       | query -> Some (Format.asprintf "(%s)" query)
     in
+    CCOption.to_list guardian |> CCString.concat " AND " |> Lwt.return
+  ;;
+
+  let query_by_admin where ?query actor pool =
+    let%lwt guardian_conditions = find_by_user_params pool actor in
+    let where =
+      let sql = Format.asprintf "%s AND %s" guardian_conditions where in
+      sql, Dynparam.empty
+    in
+    Query.collect_and_count
+      pool
+      query
+      ~select:find_request_sql
+      ~where
+      Repo_entity.t
+  ;;
+
+  let find_incomplete_by_admin =
+    {sql|
+      pool_sessions.closed_at IS NULL
+      AND pool_sessions.canceled_at IS NULL
+      AND (pool_sessions.start + INTERVAL duration SECOND) < NOW()
+    |sql}
+    |> query_by_admin
+  ;;
+
+  let find_upcoming_by_admin =
+    {sql|
+      (pool_sessions.start + INTERVAL duration SECOND) > NOW()
+      AND pool_sessions.closed_at IS NULL
```

Comment:
guess, `canceled_at IS NULL` should be in here as well

Reply:
Currently, I am displaying canceled sessions with a red background, so assistants would see if a session was canceled. We could also add a filter to hide or show them?

Reply:
Added legend, as discussed